pandas/core/frame.py

```diff
@@ -511,7 +512,7 @@ def _get_axes(N, K, index=index, columns=columns):
             index, columns = _get_axes(len(values), 1)
             return _arrays_to_mgr([values], columns, index, columns,
                                   dtype=dtype)
-        elif is_datetimetz(values):
+        elif (is_datetimetz(values) or is_extension_array_dtype(values)):
```

Comment:
```python
In [7]: pd.Series([0, 1, 2], dtype=cyberpandas.IPType())
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-7fc01964da1d> in <module>()
----> 1 pd.Series([0, 1, 2], dtype=cyberpandas.IPType())

~/sandbox/pandas-ip/pandas/pandas/core/series.py in __init__(self, data, index, dtype, name, copy, fastpath)
    246             else:
    247                 data = _sanitize_array(data, index, dtype, copy,
--> 248                                        raise_cast_failure=True)
    249
    250                 data = SingleBlockManager(data, index, fastpath=True)

~/sandbox/pandas-ip/pandas/pandas/core/series.py in _sanitize_array(data, index, dtype, copy, raise_cast_failure)
   3214         if dtype is not None:
   3215             try:
-> 3216                 subarr = _try_cast(data, False)
   3217             except Exception:
   3218                 if raise_cast_failure:  # pragma: no cover

~/sandbox/pandas-ip/pandas/pandas/core/series.py in _try_cast(arr, take_fast_path)
   3168             subarr = maybe_cast_to_datetime(arr, dtype)
   3169             if not is_extension_type(subarr):
-> 3170                 subarr = np.array(subarr, dtype=dtype, copy=copy)
   3171         except (ValueError, TypeError):
   3172             if is_categorical_dtype(dtype):

TypeError: data type not understood
```

How should we handle this? I'm fine with raising with a better error message. We don't know how to cast from `(arbitrary array, extension_dtype) -> extension array`.

Although... maybe we could support `Series(extension_array, dtype=extension_dtype)`.

Oh hey, that "works" 

```python
In [10]: pd.Series(cyberpandas.IPArray([0, 1, 2]), dtype=cyberpandas.IPType())
Out[10]:
0    0.0.0.0
1    0.0.0.1
2    0.0.0.2
dtype: ip

```

But it only works since we ignore the dtype entirely :)

I will raise if `data.dtype` is not dtype equal with `dtype` I think.

Reply:
also for things like `astype(SomeExtensionDtype)` we will need an informative error message

Reply:
isn't a dateimetz an extension type now? why can't you remove the first part of the clauase?

Reply:
Nope, not yet. Just `Categorical` so far. I have `Interval` and `Period` started, but no PRs yet.